lib/active_interaction/base.rb

```diff
@@ -214,13 +215,35 @@ def inputs
     #   Example.run!(x: nil)  # => true
     #   Example.run!(x: rand) # => true
     #
+    # @example Nested checks
+    #   class Example < ActiveInteraction::Base
+    #     hash :x, default: {} do
+    #       integer :y, default: nil
+    #     end
+    #     def execute; given?(:x, :y) end
+    #   end
+    #   Example.run!()               # => false
+    #   Example.run!(x: nil)         # => false
+    #   Example.run!(x: {})          # => false
+    #   Example.run!(x: { y: rand }) # => true
+    #
     # @param input [#to_sym]
     #
     # @return [Boolean]
     #
     # @since 2.1.0
-    def given?(input)
-      @_interaction_keys.include?(input.to_sym)
+    def given?(input, *rest)
+      filter_level = self.class
+      input_level = @_interaction_inputs
+
+      [input, *rest].map(&:to_sym).each do |key|
+        filter_level = filter_level.filters[key]
+        if !filter_level || input_level.nil? || !input_level.key?(key)
```

Comment:
This will blow up if one of the intermediate inputs doesn't respond to `#key?`, right? For example, if you ask `given?(:a, :b, :c)` and the inputs were `{ a: { b: [] } }`, I think this will explode.